report.js

```diff
@@ -3,12 +3,14 @@ const os = require('os')
 const exec = require('child_process').exec
 const localDevPort = process.env.SCARF_LOCAL_PORT
 const https = localDevPort ? require('http') : require('https')
-const fs = require('fs').promises
+const fs = require('fs')
+const fsAsync = fs.promises
 
 const scarfHost = localDevPort ? 'localhost' : 'scarf.sh'
 const scarfLibName = '@scarf/scarf'
 
 const rootPath = path.resolve(__dirname).split('node_modules')[0]
+const tmpFileName = `${os.tmpdir()}/scarf-js-history.log`
```

Comment:
Problem:

- 2 users on a system will fire an error competing for this file
- 2 builds on a system will fire an error competing for this file

Solution:

```
const os = require('os');
const path = require('path');
const UUID = require('pure-uuid');

const id = new UUID(4).format();
const tmpFileName = path.join(os.tmpdir(), "scarf-js-history-${id}.log");
```

Reply:
So, this uuid route would mean that every scarf installation would grab its own file. We'd never hit the rate limit because every process is reading from a different file every time, we'd never access the same file twice.

Two users on a system is a good point - Could be solved by adding the user id or name into that file. I'll add that in. 

Because we handle errors by simply assuming there has not been any previous logs, I think two competing builds from the same user is ok. If there is a race condition we'll will exceed the rate limit, but that's better than silently reporting analytics without telling the user at at all. 

Thoughts?

Reply:
Resolved offline